app/src/main/java/nu/yona/app/ui/YonaActivity.java

```diff
@@ -349,11 +349,12 @@ public void onResume()
 
 	private void checkForNotificationPermission()
 	{
-		if (!NotificationManagerCompat.from(this).areNotificationsEnabled())
+		if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O || (NotificationManagerCompat.from(this).areNotificationsEnabled() && !AppUtils.arePersistentNotificationsEnabled(getActivity(), AppConstant.YONA_SERVICE_CHANNEL_ID)))
```

Comment:
Is this right?
It now returns when:
* Android is older than O
* The notifications are enabled and the persistent notifications are NOT enabled

That sounds wrong to me. I would expect this:
It now returns when:
* Android is older than O
* The notifications, including our persistent one, are enabled

Reply:
Functionality wise it was working fine, problem was with the way I'm using those conditions.
I have rectified them now

Reply:
Yes, I see what you mean now.